app/assets/javascripts/application.js

```diff
@@ -351,6 +351,12 @@ $(function() {
   });
 });
 
+function check_for_unavailable_puppetclasses() {
+  if ($('#puppet_klasses #selected_classes .unavailable').size() > 0) {
+    notify('<span>' + __('Some Puppet Classes are unavailable in the selected environment') + '</span>', 'warning')
```

Comment:
I think it's 'Puppet class' (lowercase class) - we're certainly not being very consistent about this around the app but a quick search revealed it's only 'Puppet Class' on a couple of titles, not texts